src/components/Table/hooks/useSortBy.ts

```diff
@@ -0,0 +1,67 @@
+import { useCallback, useMemo, useState } from 'react';
+import type { SortingDirection } from '../types';
+
+type Sorting = {
+    direction: SortingDirection;
+    field: string;
+};
+
+interface SortBy {
+    sortBy: Sorting;
+    setSortBy: (targetField: string) => void;
+}
+
+/**
+ * This hook holds the state of the current field and the direction in which that field should be sorted.
+ * When a field is active, updating the sorting (with the same field key), will make the state cycle through "DESC" => "ASC" => and then inactive (clears the field, and `active` will turn false).
+ * The default initial sorting direction is ASC (ascending order).
+ * @param  {string} (defaultField) the name/key of the field that will be sorted by default (following BE response, most probably).
+ * @param  {string} (initialDirection) the sorting order/direction to use initially when a field is not yet being used for sorting.
+ */
+function useSortBy(defaultField: Sorting['field'] = '', initialDirection: SortingDirection = 'ASC'): SortBy {
+    const [field, setField] = useState(defaultField);
+    const [direction, setDirection] = useState<SortingDirection>(initialDirection);
+
+    /**
+     * If no field is currently being sorted, or the target field is different from the current field, the target field will become the current field and the sorting order/direction will be set to the initial direction.
+     * If the target field matches the current field being sorted, the sorting direction will toggle from DESC to ASC and vice-versa.
+     * If the sorting direction is already ASC, and the target field matches the current field, then the sorting will be reset.
+     * @param  {string} (targetField) the name/key of the field to apply sorting to.
+     */
+    const setSortBy = useCallback(
+        (targetField: string) => {
```

Comment:
Not sure about the nested ternaries...

BTW, I think there's an error:
```js
const isInitialDirection = direction == initialDirection *
```

Reply:
Would be cool to switch enum states instead of adding conditional logic here you know. 

Imagine we have 
```ts
enum Direction { ASC, DESC, NONE }
```
and we can just cycle through the states.

Reply:
I agree it can be simplified. But @lloydaf's suggestion doesn't take care of the default field case.

I'll be pushing a simplified version soon (without nested ternaries).

Reply:
I think it is better, one suggestion would be to remove the `return` and use the `else if` pattern for the next `if` case just to keep it consistent, but I think either way its more readable this way

Reply:
Done.